crates/build-rs/src/input.rs

```diff
@@ -554,13 +609,10 @@ pub fn cargo_pkg_readme() -> Option<PathBuf> {
     to_opt(var_or_panic("CARGO_PKG_README")).map(to_path)
 }
 
-fn is_present(key: &str) -> bool {
-    var_os(key).is_some()
-}
-
 #[track_caller]
 fn var_or_panic(key: &str) -> std::ffi::OsString {
```

Comment:
Changing this to e.g. `ENV.require(key)` may be a good idea to keep env access more uniform, and paves the way towards maybe supporting a different protocol in the future.

Reply:
I intentionally kept it off.

I see two ways of going
- Minimum interface for what is needed
- Every input being an inherent function on the trait

I went with the former.  I only included `is_present` on the trait because I was concerned about what the right policy is for re-run if changed in that case.  The direction I went with doesn't specialize it but I figured I'd still leave that door open.  I'm not seeing a reason to specialize this atm and would rather wait for use cases rather than speculatively adding functionality to the trait.

Since we can add this as an inherent function, we can likely do so without a breaking change (yes, there are caveats).  Even then, its not a big deal to make a breaking change.

Reply:
Especially since the trait/type aren't `pub` (yet?) and are just internal helpers.